### ISSUE TITLE
I would like to contribute to OpenWatcom, should I go ahead?

### ISSUE THREAD

**Issue by pgrAm** [open-watcom/open-watcom-v2#354]
Hi I am a very satisfied user of OpenWatcom, however the lack of C++11 support is admittedly a big downside. I would like to help towards C++11 support. I am an experienced C/C++/x86 assembly developer and use OpenWatcom mostly for 16bit code, however I don't know that much about compilers so I thought I could help in implementing the standard library for C++11, starting with implementing std::chrono as it appears to not need any new language features.

Is there anyone already planning to implementing this already, is there anything I should know besides whats already on the wiki before diving in?

**Comment by bmanga** [comment]
Hi @pgrAm, contributions are indeed very welcome. Chrono is a good choice because as you said it doesn't rely on too many c++11 features. It is however fairly platform-dependent, and you cannot just piggyback on ctime to get decent results.
If you are up for it, go ahead :)

On the lack of c++11 features, I have just started the process of rewriting parts of the template code. This should eventually unlock progress on parameter packs, forwarding references, auto type deduction (and hence range-based for loop), alias templates etc.
I am not sure how long it will take though.

**Comment by pgrAm** [comment]
Great to hear about the progress on C++11, I think I'll take a crack at Chrono, I should be able to implement most of the platform specific stuff for the platforms I work on (Windows, DOS, Linux) with no trouble, then I'll get started on the other ones.

**Comment by jmalak** [comment]
Hi both.
I think most important is to change names processing in C ++ compiler before any other work.
In OW C++ compiler it is somehow fixed for Class Name (one level only) and member name.
It is main limitation for full template support.

**Comment by bmanga** [comment]
@jmalak are you able to work on that?

**Comment by jmalak** [comment]
I am not much experienced with C++ code.  
I don't feel as right person to do this, it needs high level of C++ abstraction for names processing.
I can help you with some knowledge about compiler and code generator and OW at all.

**Comment by bmanga** [comment]
I opened a separate issue #355

**Comment by pchapin** [comment]
When adding new facilities to the C++ library, don't forget to add tests in plustest/regress.
